devtools/datasette/publish.py

```diff
@@ -82,9 +83,78 @@ def inspect_data(datasets: list[str], pudl_out: Path) -> str:
     return inspect_output
 
 
-def metadata(pudl_out: Path) -> str:
+def check_tables_have_metadata(
+    metadata: DatasetteMetadata,
+    pudl_out: Path,
+    datasets: list[str],
+) -> None:
+    """Check to make sure all tables in the databases have metadata.
+
+    This function fails if there are tables present in a database
+    that do not have structure metadata.
+
+    Args:
+        metadata: The structure metadata for the datasette deployment
+        pudl_out: The directory that contains the pudl outputs
+        datasets: The list of datasets to test.
```

Comment:
* What is a "dataset" here? What set of strings are you referring to? Is this like `eia923` and `epacems` or something else?
* Can we not use `pudl_out` for anything else for a while, until the memory of it has faded?

Reply:
- Yes! "dataset" is confusing. Will update to databases.
- What do you mean by `pudl_out`? Are we trying to remove it?

Reply:
Ah sorry about the confusion.

For years, we have used `pudl_out` as the name for a `PudlTabl` output object that's passed around everywhere, so re-using that name while `PudlTabl` is still floating around feels confusing/distracting to me. So I would recommend calling it `pudl_output` instead, which is the same as the `$PUDL_OUTPUT` environment variable which typically defines it now.

Reply:
Oh I see. Thanks!